### PR TITLE
Resolved conflicts for MergeCells plugin

### PR DESCRIPTION
**Pull request by wszymanski**
### Context
PR #10517 has been merged to epic branch instead of develop. This PR just resolves conflicts.

[skip changelog]

### CODE REVIEW

File: handsontable/src/plugins/copyPaste/clipboardData/clipboardData.js [hunk 1]
```diff
@@ -0,0 +1,546 @@
...
+   *
+   * @type {string}
+   */
+  html;
+  /**
+   * Copied data stored as array of arrays.
+   *
+   * @type {string[][]}
+   */
+  data;
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/clipboardData/clipboardData.js:33]:
```suggestion
  /**
   * Sanitized data of "text/html" type inside the clipboard.
   *
   * @private
   * @type {string}
   */
  html;
  /**
   * Copied data stored as array of arrays.
   *
   * @private
   * @type {string[][]}
   */
  data;
```

File: handsontable/src/plugins/copyPaste/clipboardData/copyClipboardData.js [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+   * @private
+   * @type {string[][]}
+   */
+  data;
+  /**
+   * Copied cell ranges related to instance of Handsontable.
+   *
+   * @type {Array<{startRow: number, startCol: number, endRow: number, endCol: number}>}
+   */
+  copyableRanges;
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/clipboardData/copyClipboardData.js:30]:
```suggestion
  /**
   * Copied cell ranges related to instance of Handsontable.
   *
   * @private
   * @type {Array<{startRow: number, startCol: number, endRow: number, endCol: number}>}
   */
  copyableRanges;
```

File: handsontable/src/plugins/copyPaste/clipboardData/pasteClipboardData.js [hunk 3]
```diff
@@ -0,0 +1,78 @@
...
+  getType() {
+    if (this.isSerializedHandsontable()) {
+      return 'handsontable';
+    }
+
+    if (this.isSerializedTable()) {
+      return 'table';
+    }
+
+    return 'string';
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/clipboardData/pasteClipboardData.js:76]: Wouldn't it be better to name the unknown type (unknown type to parse) as `unknown` (or `unrecognizable`)?

File: handsontable/src/plugins/copyPaste/clipboardData/pasteClipboardData.js [hunk 4]
```diff
@@ -0,0 +1,78 @@
...
+   * @returns {boolean}
+   */
+  isSerializedHandsontable() {
+    return this.isSerializedTable() && /<meta (.*?)content="Handsontable"/.test(this.html);
+  }
+
+  /**
+   * Gets type of the copied data.
+   *
+   * @returns {string}
```
Comment (budnix) [inline handsontable/src/plugins/copyPaste/clipboardData/pasteClipboardData.js:65]:
```suggestion
   * @returns {'table' | 'handsontable' | 'unknown'}
```

File: handsontable/types/pluginHooks.d.ts [hunk 5]
```diff
@@ -49,13 +49,35 @@ interface HookHighlightRowHeaderMeta {
...
   selectionWidth: number;
   classNames: string[];
 }
 
+interface ClipboardData {
+  removeRows: (rows: number[]) => void;
+  removeColumns: (columns: number[]) => void;
+  insertAtRow: (rowIndex: number, values: any[]) => void;
+  insertAtColumn: (columnIndex: number, values: any[]) => void;
+  getCellAt: (row: number, col: number) => 'string';
```
Comment (budnix) [inline handsontable/types/pluginHooks.d.ts:65]:
```suggestion
  getCellAt: (row: number, col: number) => string;
```

File: handsontable/types/pluginHooks.d.ts [hunk 6]
```diff
@@ -49,13 +49,35 @@ interface HookHighlightRowHeaderMeta {
...
+  setCellAt: (row: number, col: number, value: any) => void;
```
Comment (budnix) [inline handsontable/types/pluginHooks.d.ts:66]:
```suggestion
  setCellAt: (row: number, column: number, value: any) => void;
```

File: handsontable/types/pluginHooks.d.ts [hunk 7]
```diff
@@ -49,13 +49,35 @@ interface HookHighlightRowHeaderMeta {
...
+  getData: () => any[][];
+  getMetaInfo: () => GridSettings;
+  getType: () => string;
+}
+
+interface PasteClipboardData extends ClipboardData {
+  getType: () => 'handsontable' | 'table' | 'string';
```
Comment (budnix) [inline handsontable/types/pluginHooks.d.ts:73]:
```suggestion
  getType: () => 'handsontable' | 'table' | 'unknown';
```

**Review by budnix — CHANGES_REQUESTED** [review]
I left some small, non-time-consuming suggestions to consider.

File: handsontable/src/plugins/copyPaste/clipboardData/pasteClipboardData.js (hunk 3 above)
Comment (wszymanski) [reply handsontable/src/plugins/copyPaste/clipboardData/pasteClipboardData.js]: It has been done. Thank you.

File: handsontable/types/pluginHooks.d.ts [hunk 8]
```diff
@@ -49,13 +49,35 @@ interface HookHighlightRowHeaderMeta {
...
+interface ClipboardData {
+  removeRows: (rows: number[]) => void;
+  removeColumns: (columns: number[]) => void;
+  insertAtRow: (rowIndex: number, values: any[]) => void;
+  insertAtColumn: (columnIndex: number, values: any[]) => void;
+  getCellAt: (row: number, col: number) => string;
+  setCellAt: (row: number, column: number, value: any) => void;
+  getData: () => any[][];
+  getMetaInfo: () => GridSettings;
+  getType: () => string;
```
Comment (budnix) [inline handsontable/types/pluginHooks.d.ts:69]:
```suggestion
  getType: () => 'handsontable ';
```

**Review by budnix — APPROVED** [review]

**Pull request merged**